akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -159,6 +165,48 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
         )
       }
     }
+
+    "respect settings" should {
+      "received SETTINGS_MAX_CONCURRENT_STREAMS should limit the number of outgoing streams" in new TestSetup(
+        Setting(SettingIdentifier.SETTINGS_MAX_CONCURRENT_STREAMS, 3)
+      ) with NetProbes {
+        val request = HttpRequest(uri = "https://www.example.com/")
+        // server set SETTINGS_MAX_CONCURRENT_STREAMS=5 so an attempt from the client to open 6 streams
+        // should only produce 5 frames
+        emitRequest(1, request)
+        emitRequest(3, request)
+        emitRequest(5, request)
+        emitRequest(7, request) // this is backpressured
```

Comment:
Shouldn't this fail then already? Or do the probes introduce a bit of buffering?